model/measurement.py

```diff
@@ -35,36 +35,68 @@ class Measurement(Base):
 
-    # If a book's popularity measurement is found between index n and
-    # index n+1 on this list, it is in the nth percentile for
-    # popularity and its 'popularity' value should be n * 0.01.
+    # We have a number of ways of measuring popularity: by an opaque
+    # number such as Amazon's Sales Rank, or by a directly measured
+    # quantity such as the number of downloads or published editions,
+    # or the number of libraries with a given book in their
+    # collection.
     #
-    # These values are empirically determined and may change over
-    # time.
-    POPULARITY_PERCENTILES = {
-        DataSourceConstants.OVERDRIVE : [1, 1, 1, 2, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 7, 7, 8, 9, 9, 10, 10, 11, 12, 13, 14, 15, 15, 16, 18, 19, 20, 21, 22, 24, 25, 26, 28, 30, 31, 33, 35, 37, 39, 41, 43, 46, 48, 51, 53, 56, 59, 63, 66, 70, 74, 78, 82, 87, 92, 97, 102, 108, 115, 121, 128, 135, 142, 150, 159, 168, 179, 190, 202, 216, 230, 245, 260, 277, 297, 319, 346, 372, 402, 436, 478, 521, 575, 632, 702, 777, 861, 965, 1100, 1248, 1428, 1665, 2020, 2560, 3535, 5805],
-        DataSourceConstants.AMAZON : [14937330, 1974074, 1702163, 1553600, 1432635, 1327323, 1251089, 1184878, 1131998, 1075720, 1024272, 978514, 937726, 898606, 868506, 837523, 799879, 770211, 743194, 718052, 693932, 668030, 647121, 627642, 609399, 591843, 575970, 559942, 540713, 524397, 511183, 497576, 483884, 470850, 458438, 444475, 432528, 420088, 408785, 398420, 387895, 377244, 366837, 355406, 344288, 333747, 324280, 315002, 305918, 296420, 288522, 279185, 270824, 262801, 253865, 246224, 238239, 230537, 222611, 215989, 208641, 202597, 195817, 188939, 181095, 173967, 166058, 160032, 153526, 146706, 139981, 133348, 126689, 119201, 112447, 106795, 101250, 96534, 91052, 85837, 80619, 75292, 69957, 65075, 59901, 55616, 51624, 47598, 43645, 39403, 35645, 31795, 27990, 24496, 20780, 17740, 14102, 10498, 7090, 3861],
-
-        # This is as measured by the criteria defined in
-        # ContentCafeSOAPClient.estimate_popularity(), in which
-        # popularity is the maximum of a) the largest number of books
-        # ordered in a single month within the last year, or b)
-        # one-half the largest number of books ever ordered in a
-        # single month.
-        DataSourceConstants.CONTENT_CAFE : [0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 3, 3, 3, 3, 3, 3, 3, 3, 3, 4, 4, 4, 4, 4, 4, 5, 5, 5, 5, 6, 6, 7, 8, 9, 10, 11, 14, 18, 25, 41, 125, 387]
-
-        # This is a percentile list of OCLC Work IDs and OCLC Numbers
-        # associated with Project Gutenberg texts via OCLC Linked
-        # Data.
+    # All of these ways of measuring popularity need to be scaled to a
+    # range between 0 and 1. This nested dictionary contains
+    # empirically determined measurements keyed by what they are measureing
```

Comment:
I don't think I've ever looked at this code closely and now I'm curious how these were determined.

Reply:
I take N measurements, lay them out in a sorted list, and take evenly spaced chunks from the list.

For OCLC Classify, N ~= 35,000 recently published books.